plugins/woocommerce/includes/class-wc-order-item-shipping.php

```diff
@@ -122,6 +123,19 @@ protected function set_total_tax( $value ) {
 		$this->set_prop( 'total_tax', wc_format_decimal( $value ) );
 	}
 
+	/**
+	 * Set tax_status.
+	 *
+	 * @param string $value Tax status.
+	 */
+	public function set_tax_status( $value ) {
+		if ( in_array( $value, array( 'taxable', 'none' ), true ) ) {
+			$this->set_prop( 'tax_status', $value );
+		} else {
+			$this->set_prop( 'tax_status', 'taxable' );
+		}
+	}
```

Comment:
I find it a bit weird that calling `set_tax_status()` with an incorrect value will set the property to "taxable" rather than throw an error or simply do nothing.
If this is done so that initialization in the datastore is easier, maybe we should sanitize the value there before calling `set_tax_status()` on the object. Thoughts?

Reply:
To be frank I followed what other line items were doing here in fear of breaking something. I think we can remove the else part and default to taxable.